### DISCUSSION TITLE
7250 IXR-e close to working

### DISCUSSION THREAD

**Discussion by rp-nz** [hellt/vrnetlab#8] [Ideas]
Ok the issue described below is now fixed:
Found the issue why the line card was failing to boot. It it is multicore (smp) line card and requires at least 2 vcpus to boot.
Edited the vrnetlab.py file and added following, -cpu host and the important one -smp 2. 
```
       self.qemu_args.extend(
            [
                "-m",
                str(ram),
                "-serial",
                "telnet:0.0.0.0:50%02d,server,nowait" % self.num,
                "-drive",
                "if=ide,file=%s" % overlay_disk_image,
                "-cpu",
                "host",
                "-smp",
                "2",
            ]
        )
```

Hi Roman,
Nice work on getting support for the recent version of the Nokia VSIMs within vrnetlab. 
I taken the latest copy (your fork) of the vrnetlab and as shown below I have tried to add the Nokia 7250 IXR-e variant. 
While the 7250 IXR-e is a fixed form factor node but the VSIM requires two VMs, i.e. distributed mode of operation. 
I have this working on eve-ng using a CPM and IMM based VMs. 
Based on the changes below the VMs boot with docker but I get following message in the CPM VM:
As you can see the card has failed.
I am not sure why this is when what i can configured is same as what i have done in eve-ng. 

I can see that in the launch.py it provides the option for setting timos specific options, but where do pass additional flags for qemu. the reason I asked about this is when i look at my eve-ng, i can see following flags for qemu:
`-cpu host` as per following in eve-ng which works fine:

**IXR-e CPM in eve-ng:** 
`-machine type=pc,accel=kvm -enable-kvm -uuid <uuid value> -serial mon:stdio -nographic -nodefaults -rtc base=utc -cpu host -smbios type=1,product="TIMOS:slot=A chassis=IXR-e card=cpm-ixr-e" 
`
**IXR-e IOM in eve-ng:**
`-machine type=pc,accel=kvm -enable-kvm -uuid <uuid value> -serial mon:stdio -nographic -nodefaults -rtc base=utc -cpu host -smbios type=1,product="TIMOS:slot=1 chassis=IXR-e card=imm24-sfp++8-sfp28+2-qsfp28 mda/1=m24-sfp++8-sfp28+2-qsfp28" `

Can you clarify one thing for me please, does the VMs run inside the docker container, because on my host I see four processes of qemu-system running when I started the 7250 IXR-e containers using docker, so I wondering where is the VMs running, on the host or within the docker container?  

Card fails in the vrnetlab version of booting IXR-e. 
```
A:admin@vr-sros# show card

===============================================================================
Card Summary
===============================================================================
Slot      Provisioned Type                         Admin Operational   Comments
              Equipped Type (if different)         State State         
-------------------------------------------------------------------------------
1         imm24-sfp++8-sfp28+2-qsfp28              up    failed        
A         cpm-ixr-e                                up    up/active     
===============================================================================

[]

```
Here are the changes I made to the lanuch.py file and recreated the docker image. 

```
    "sr-1": {
        "deployment_model": "integrated",
        "min_ram": 5120,  # minimum RAM requirements
        "max_nics": 6,
        "timos_line": "chassis=sr-1 slot=A card=cpm-1 slot=1 mda/1=me6-100gb-qsfp28",
        "card_config": """/configure card 1 card-type iom-1
        /configure card 1 mda 1 mda-type me6-100gb-qsfp28
        """,
    },
    "sr-1e": {
        "deployment_model": "distributed",
        # control plane (CPM)
        "max_nics": 20,
        "cp": {
            "min_ram": 4096,
            "timos_line": "slot=A chassis=sr-1e card=cpm-e",
        },
        # line card (IOM/XCM)
        "lc": {
            "min_ram": 4096,
            "timos_line": "chassis=sr-1e slot=1 card=iom-e mda/1=me40-1gb-csfp",
            "card_config": """/configure card 1 card-type iom-e
            /configure card 1 mda 1 mda-type me40-1gb-csfp
            /configure card 1 fp 1
            """,
        },
    },
    "ixr-e": {
        "deployment_model": "distributed",
        # control plane (CPM)
        "max_nics": 20,
        "cp": {
            "min_ram": 6048,
            "timos_line": "slot=A chassis=IXR-e card=cpm-ixr-e",
        },
        # line card (IOM/XCM)
        "lc": {
            "min_ram": 6096,
            "timos_line": "chassis=IXR-e slot=1 card=imm24-sfp++8-sfp28+2-qsfp28 mda/1=m24-sfp++8-sfp28+2-qsfp28",
            "card_config": """/configure card 1 card-type imm24-sfp++8-sfp28+2-qsfp28
            /configure card 1 mda 1 mda-type m24-sfp++8-sfp28+2-qsfp28
            """,
        },
    },
}
```
`choices=["sr-1", "sr-1e", "ixr-e"],`

**Comment by hellt** [comment]
oh, did I miss that? It seems the issue has been addressed already...
Sorry for not providing a response, it avoided my mailbox
